src/scripts/main.js

```diff
@@ -1,3 +1,35 @@
 'use strict';
 
 // write code here
+const tree = document.querySelector('.tree');
+
+(function wrapHeaderTextInSpan() {
+  const listOfLi = [...document.querySelectorAll('li')];
+
+  listOfLi.forEach((li) => {
+    const textNode = li.firstChild;
+
+    if (
+      textNode &&
+      textNode.nodeType === Node.TEXT_NODE &&
+      li.querySelector('ul')
+    ) {
+      const span = document.createElement('span');
+
+      span.textContent = textNode.textContent.trim();
+      textNode.replaceWith(span);
+    }
+  });
+})();
+
+tree.addEventListener('click', (e) => {
```

Comment:
Ensure that the 'tree' element is not `null` before adding an event listener to it. If 'tree' is `null`, this will throw an error.